src/app/public/testing/summary-action-bar/summary-action-bar-fixture.spec.ts

```diff
@@ -0,0 +1,284 @@
+
+import {
+  SkyKeyInfoModule
+} from '@skyux/indicators';
+
+import {
+  SummaryActionBarTestComponent
+} from './fixtures/summary-action-bar-fixture-test.component';
+
+import {
+  SkySummaryActionBarFixture
+} from './summary-action-bar-fixture';
+
+import {
+  SkySummaryActionBarTestingModule
+} from './summary-action-bar.module';
+
+describe('Summary action bar fixture', () => {
+  let fixture: ComponentFixture<SummaryActionBarTestComponent>;
+  let testComponent: SummaryActionBarTestComponent;
+  let mockQueryService: MockSkyMediaQueryService;
+  let summaryActionBarFixture: SkySummaryActionBarFixture;
+
+  //#region helpers
+
+  async function initiateResponsiveMode(breakpoint: SkyMediaBreakpoints): Promise<void> {
+    mockQueryService.fire(breakpoint);
+    fixture.detectChanges();
+    return fixture.whenStable();
+  }
+
+  //#endregion
+
+  beforeEach(() => {
+    // replace the mock service before using in the test bed to avoid change detection errors
+    mockQueryService = new MockSkyMediaQueryService();
+
+    TestBed.configureTestingModule({
+      declarations: [
+        SummaryActionBarTestComponent
+      ],
+      imports: [
+        SkyKeyInfoModule,
+        SkySummaryActionBarTestingModule
+      ],
+      providers: [
+        { provide: SkyMediaQueryService, useValue: mockQueryService }
+      ]
+    });
+
+    fixture = TestBed.createComponent(
+      SummaryActionBarTestComponent
+    );
+    testComponent = fixture.componentInstance;
+    fixture.detectChanges();
+    summaryActionBarFixture = new SkySummaryActionBarFixture(
+      fixture,
+      SummaryActionBarTestComponent.dataSkyId
+    );
+  });
+
+  describe('primary action', () => {
+
+    it('should expose primary action button default properties', async () => {
+      const actionSpy = spyOn(fixture.componentInstance, 'primaryActionClicked');
+      const action = summaryActionBarFixture.primaryAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose primary action button default properties when responsive', async () => {
+      await initiateResponsiveMode(SkyMediaBreakpoints.xs);
+      const actionSpy = spyOn(fixture.componentInstance, 'primaryActionClicked');
+      const action = summaryActionBarFixture.primaryAction;
+
+      // verify button state
+      expect(action.buttonText).toEqual(testComponent.primaryAction.buttonText);
+      expect(action.isDisabled).toEqual(testComponent.primaryAction.isDisabled);
+
+      // verify the click methods work on the button state
+      await action.click();
+      expect(actionSpy).toHaveBeenCalled();
+    });
+
+    it('should expose primary action button properties', async () => {
```

Comment:
Could we say `should expose primary action button properties if the properties change`? That seems to be what we are testing. I just want to be clear when looking down the road about what we are testing compared to the firs test.